docs/howtos/upgrading-nss-guide.md

```diff
@@ -0,0 +1,20 @@
+# Guide to upgrading NSS
+
+Our components rely on cryptographic primitives provided by [NSS](https://developer.mozilla.org/docs/Mozilla/Projects/NSS).
+Every month or so, a new version of NSS is [published](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSS/NSS_Releases) and we should try to keep our version as up-to-date as possible.
+
+For various reasons, we compile NSS ourselves and link to it statically. Note that NSS is mainly used by Mozilla as a dynamic library and the NSS project is missing related CI jobs (iOS builds, windows cross-compile builds etc.) so you should expect breakage when updating the library (hence this guide).
+
+The build code is located in the [`libs/`](https://github.com/mozilla/application-services/tree/main/libs) folder.  
+The version string is located in the beggining of [`build-all.sh`](https://github.com/mozilla/application-services/blob/b0b3daa6580d04906fc53e9e479e8bebb464cf78/libs/build-all.sh#L8-L11): for most NSS upgrades, the only action needed is to bump the version number in this file and update the downloaded archive checksum.  Note that each platform has its own nss build script (e.g. [`build-nss-ios.sh`](https://github.com/mozilla/application-services/blob/b0b3daa6580d04906fc53e9e479e8bebb464cf78/libs/build-nss-ios.sh))
```

Comment:
sp: "beginning".

Reply:
nit: The colon at the end of the "build-all.sh" link text doesn't parse for me, should this be a plain full-stop?

Reply:
nit: "nss" -> "NSS" for consistency in "has its own nss build script".

Reply:
As a reader, it's not clear to me what I'm supposed to do with the knowledge that "each platform has its own nss build script". Do you mean something like "...which you might need to update on occasion if/when it breaks"?